Unicode UI/MitSciUnicodeUI.pax

```diff
@@ -0,0 +1,1283 @@
+!ListView class categoriesFor: #initializeNotificationMap!initializing!must not strip!private! !
+
+!Menu methodsFor!
+
+userLibrary
+
+	"Private - cdemers 10/26/2016"
+	"cdemers 11/23/2016 Revised to use the class of self text to determin the UserLibrary to use."  
+	^((self text class == UnicodeString ) ifTrue: [UserLibraryW] ifFalse: [UserLibrary]) default! !
+!Menu categoriesFor: #userLibrary!private! !
+
+!MenuItem methodsFor!
+
+userLibrary
+
+	"Private - cdemers 10/26/2016"
+	"cdemers 11/23/2016 Revised to use the class of self text to determin the UserLibrary to use."  
+	^((self text class == UnicodeString ) ifTrue: [UserLibraryW] ifFalse: [UserLibrary]) default! !
+!MenuItem categoriesFor: #userLibrary!private!realizing/unrealizing! !
+
+!MessageBox methodsFor!
+
+userLibrary
+
+	"cdemers 12/1/2017 Retrun either the UserLibrary with either ANSI or Unicode support based on the kind of text string."
+	^((self text class == UnicodeString ) ifTrue: [UserLibraryW] ifFalse: [UserLibrary]) default! !
+!MessageBox categoriesFor: #userLibrary!public! !
+
+!Presenter class methodsFor!
+
+isUnicodeMode
+
+	"cdemers 12/2/2016 - Provide an event hook for translation managers to devide if we should open in Unicode mode."
+	| boolValue |
+
+	boolValue := self -> false.
+	self trigger: #isUnicodeRequested: with: boolValue.
+	^boolValue value!
+
+loadViewResource: aString inContext: aParentView 
+	"cdemers 10/20/2016 Use a creation block to set Unicode status."
+
+	| resourceIdentifier  newView |
+
+	self isUnicodeMode 
+		ifTrue: [resourceIdentifier := ResourceIdentifier class: self name: aString.
+			ShellView createHookBlock: 
+					[:shellView :createBlock | 
+					"| exStyle |
+					exStyle := shellView extendedStyle."
+					shellView isUnicodeView: true.
+					createBlock value.
+					"shellView extendedStyle: exStyle"].
+			[newView  := resourceIdentifier loadWithContext: aParentView] ensure: 
+				[ShellView createHookBlock: nil].
+			^newView]
+		ifFalse: [resourceIdentifier := ResourceIdentifier class: self name: aString.
+			^resourceIdentifier loadWithContext: aParentView]! !
+!Presenter class categoriesFor: #isUnicodeMode!public! !
+!Presenter class categoriesFor: #loadViewResource:inContext:!public! !
+
+!Prompter class methodsFor!
+
+isUnicodeMode
+
+	"cdemers 11/30/2016 decide the unicode type based on the active window."
+	^View active isUnicodeView. ! !
+!Prompter class categoriesFor: #isUnicodeMode!public! !
+
+!SymbolStringSearchPolicy methodsFor!
+
+compare: operand1 with: operand2
+	"Answer whether the <Object>, operand1, is considered equivalent to the <Object> argument,
+	operand2, by this search policy."
+	"cdemers 11/23/2016 Revised for Unicode support.  This may be innefficient."
+
+	^(operand1 asString trueCompare: operand2 asString) == 0
+!
+
+hash: operand
+	"Answer a suitable hash value for the <Object>, operand, under this search policy."
+	"cdemers 11/23/2016 Revised to accomodate UnicodeStrings"
+
+	^operand hash! !
```

Comment:
I think this will break comparison between strings and symbols when using this with a hashed collection because the hash calculated for a Symbol will be its identity hash, and therefore no longer equal to the hash of an equivalent string (which is a hash of its characters).